Checklist MFD Improvements

ThePuzzlemaker opened the pull request:
This PR adds a few helpful features to the Checklist MFD (N.B. cursor here refers to the currently selected step, but not necessarily the active step):
- When in a checklist, and the cursor is at the active step, an "UNDO" button is present which will undo the last completed step. If the active step is the first in the group, it will return to the previous checklist or to the "no active checklist" screen. No actions will be executed even if EXEC is enabled.
- When in a checklist, and the cursor is *not* at the active step, a "GOTO" button is present which will allow the user to skip forward or back to the step under the cursor, skipping any steps in between and not executing any actions.
- When in a checklist, and the cursor is *not* at the active step, a "BCK" button is present which will allow the user to sync the cursor with the active step. This is useful when perusing a checklist using PgDn and PgUp quite a bit.

ThePuzzlemaker committed:
Add "Skip" button to Checklist MFD

I have done some basic testing with this. It will skip over any sub-
checklists (e.g. P52s) and across groups, and *should* behave well in
most cases.

ThePuzzlemaker commented:
I just tested this with exec enabled, and it works--i.e., it does not flip switches at all while skipping, which is the intended behaviour.

rcflyinghokie commented:
Tested this with a few scenarios, either I am missing something or the skip button doesn't show up at all.

ThePuzzlemaker committed:
"Skip" -> "Goto" and allow reversing; "Undo" btn

This commit changes "skip" to "goto", allowing it to move both forward
and backwards in a checklist. Additionally, an "undo" button was added
that appears when the cursor is on the current item.

ThePuzzlemaker committed:
"Back" button when cursor != active step

This commit adds a "BCK" button to the checklist step navigation when
the cursor position isn't at the active step, which returns the cursor
to the active step. This is useful when using "PgDn" or "PgUp".

maxq519 reviewed (approved):
Looks good to me

rcflyinghokie reviewed (approved):
I do like the additional functionality here, going back to previous spots is great. I would encourage documenting this a bit though explaining what the buttons do just to be safe. Especially important since orbiter MFD keyboard commands depending on which side the user has it set to will press those buttons.

But testing it the functionality looks good here.
